examples/player/simid_player.js

```diff
@@ -609,14 +595,82 @@ class SimidPlayer {
     console.log('The creative has asked for the player to ping ' + requestedUrlArray);
   }
 
+  /**
+   * Called when video playback is complete.
+   * @private
+   */
+  videoComplete() {
+    this.simidProtocol.sendMessage(MediaMessage.ENDED);
+
+    if (this.requestedDuration_ == NO_REQUESTED_DURATION) {
+      this.stopAd(StopCode.MEDIA_PLAYBACK_COMPLETE);
+    }
+  }
+
+  /**
+   * Called when creative requests a change in duration of ad.
+   * @private
+   */
   onRequestChangeAdDuration(incomingMessage) {
-    if (this.requestedDuration_ != NO_REQUESTED_DURATION) {
-      // TODO: Support multiple change duration requests.
-      this.simidProtocol.reject(incomingMessage);
+    this.requestedDuration_  = incomingMessage.args['duration'];
+    if (this.requestedDuration_ != UNLIMITED_DURATION && this.requestedDuration_< 0) {
+      const durationErrorMessage = {
+        errorCode: PlayerErrorCode.UNSUPPORTED_TIME,
+        message: 'A negative duration is not valid.'
+      }
+      this.simidProtocol.reject(incomingMessage, durationErrorMessage);
+    } else if (this.requestedDuration_ == UNLIMITED_DURATION) {
+      //Note: User can x out of the ad using the close ad button on the player
+      return;
+    } else {
+      //If requested duration is any other acceptable value
+      this.changeAdEndTime();
+      this.simidProtocol.resolve(incomingMessage);
+    }
+  }
+
+  /**
+   * Compares the requested change duration with the current ad time
+   *  to determine when to stop the ad, every 250 milliseconds.
+   * @private
+   */
+  changeAdEndTime() {
+    this.interval_ = setInterval(() => {
+        this.compareAdAndRequestedDurations(); 
+    }, INTERVAL_TIMER);
+  }
+
+  /**
+   * Compares the duration of the ad with the requested change duration.
+   * If request duration is shorter, the ad stops early. 
+   * If the request duration is longer, the ad extends for the requested
+   *  amount of time.
+   * @private
+   */
+  compareAdAndRequestedDurations() {
+    const durationChangeMs = (this.requestedDuration_ - this.adVideoElement_.duration) * 1000;
+
+    if (this.adVideoElement_.ended) {
+      //If the video ad has ended already
+      setTimeout(() => {
+        this.stopAd(StopCode.CREATIVE_INITIATED);
+      }, durationChangeMs);
+      clearInterval(this.interval_);
+      return;
+    } else if (this.adVideoElement_.currentTime >= this.requestedDuration_) {
+      //Creative requested a duration shorter than the ad
+      this.stopAd(StopCode.CREATIVE_INITATED);
+      clearInterval(this.interval_);
+      return;
+    } else if (this.requestedDuration_ >= this.adVideoElement_.duration) {
+      //Creative requested a longer duration
```

Comment:
Let's make this comment meaningful to a first time reader.

// Creative requested a duration longer than the video, this will cause the ad to extend after when the video completes.